python/dolfin/function/functionspace.py

```diff
@@ -5,43 +5,50 @@
 #
 # SPDX-License-Identifier:    LGPL-3.0-or-later
 
+import typing
+
 import ufl
 from dolfin import cpp, jit
 from dolfin.fem import dofmap
 
 
+class ElementDef(typing.NamedTuple):
+    """Data for representing a finite element"""
+    family: str
+    degree: int
+    form_degree: typing.Optional[int] = None  # noqa
```

Comment:
Doesn't seem complete without a cell type? I the cell_type was `None`, FunctionSpace could infer it from the mesh?

Reply:
It's the necessary meta-data.  If the element is fully defined, one case just pass an element.